scheduler/core/programprovider/ocs/ocsprogramprovider.py

```diff
@@ -681,37 +688,40 @@ def autocorr_lag(x):
         # print(f'\t\t\t do_not_split: {do_not_split}')
 
         # all atoms must have the same instrument
-        instrument = self._parse_instrument(sequence[0])
+        instrument = self._parse_instrument(sequence)
         # print(f'instrument {instrument}')
 
         for step in sequence:
-            # Instrument configuration aka Resource.
-            fpu, disperser, filt, wavelength = OcsProgramProvider._parse_instrument_configuration(step, instrument)
-
-            # If FPU is None, 'None', or FPU_NONE, which are effectively the same thing, we ignore.
-            if fpu is not None and fpu != 'None' and fpu != 'FPU_NONE':
-                fpus.append(fpu)
-            dispersers.append(disperser)
-            if filt and filt != 'None':
-                filters.append(filt)
-            wavelengths.append(wavelength)
-
-            p = 0.0
-            q = 0.0
-
-            # Exposures on sky for dither pattern analysis
-            if step[OcsProgramProvider._AtomKeys.OBSERVE_TYPE].upper() not in OcsProgramProvider._OBSERVE_TYPES:
-                p = (float(step[OcsProgramProvider._AtomKeys.OFFSET_P]) if OcsProgramProvider._AtomKeys.OFFSET_P in step
-                     else 0.0)
-                q = (float(step[OcsProgramProvider._AtomKeys.OFFSET_Q]) if OcsProgramProvider._AtomKeys.OFFSET_Q in step
-                     else 0.0)
-                sky_p_offsets.append(p)
-                sky_q_offsets.append(q)
-            coadds.append(int(step[OcsProgramProvider._AtomKeys.COADDS])
-                          if OcsProgramProvider._AtomKeys.COADDS in step else 1)
-            exposure_times.append(step[OcsProgramProvider._AtomKeys.EXPOSURE_TIME])
-            p_offsets.append(p)
-            q_offsets.append(q)
+            # Don't consider acq steps
+            if step[OcsProgramProvider._AtomKeys.OBS_CLASS].upper() not in [ObservationClass.ACQ.name,
+                                                                            ObservationClass.ACQCAL.name]:
```

Comment:
It might be easier to just do:
```python
if step[OcsProgramProvider._AtomKeys.OBS_CLASS].upper() in [ObservationClass.ACQ.name, ObservationClass.ACQCAL.name]:
    continue
```

to avoid heavy nested if statements. Also, since we've reused `[ObservationClass.ACQ.name, ObservationClass.ACQCAL.name]` a couple times now, it would be better practice for us to add this at the top of the class as a static member like:

```python
acquisitions = frozenset([ObservationClass.ACQ.name, ObservationClass.ACQCAL.name])
```

or make a private method:
```python
def _check_step_is_acquisition(step: dict) -> bool:
    return step[OcsProgramProvider._AtomKeys.OBS_CLASS].upper() in [ObservationClass.ACQ.name, ObservationClass.ACQCAL.name]
```